### PR TITLE
fix: use usage_key instead of block_id as location identifier for scorm data

### LINKED ISSUE #70: Importing a course with scorm and changing the scorm file deletes the exported course scorm file as well
**Issue**

Exporting a course that contains a unit with a scorm module and then importing it into another course works perfectly fine. However, if you change the scorm file in either the exported or the imported course, the other one breaks with the message: 

image

This issue only happens if the exported course and the imported course are in the same openedx instance.

This error is also seen when you duplicate a subsection/unit that contains a scorm module with a scorm file already uploaded. For that, just duplicated the unit throws that error.

**Steps to reproduce**

1. Install Tutor and make sure scorm is installed by running: `tutor [dev|local] run lms pip list | grep -i scorm`
2. Create a new course.
3. Add scorm into the "Advanced Module List" under advanced settings.
4. Create a new unit with a scorm module and upload any scorm file in it.

And then after this, follow the respective steps for each issue.

To reproduce the course import/export and change scorm issue: 

1. Publish the unit and export the current course.
2. Create a new course and import the exported course into it.
3. Change the scorm file with another file (not the same scorm file) in either the exported course or the imported course.
4. The scorm module in the other course (the one in which the scorm file was not changed) will show that error.

To reproduce the subsection/duplicate issue:

1. Duplicate the unit or subsection in which the scorm module was opened.
2. Open the scorm module unit and it will show that error.

**Possible Cause**

For the course export import issue:
- Scorm saves the zip_file content at the location `scorm/block_id/hash_of_file`. It seems that the imported course keeps the same `block_id` and `hash_of_file` of the exported course and thus they reading the data from the same directory. When the scorm file is changed in either course, it overwrites the previous directory as the hash changes and the other course cannot access the data.

For the subsection/unit duplicate issue:
- Scorm only saves the contents inside the zip_file after submitting inside studio. This means that whenever the unit it duplicated or a new course is created, the data for that scorm unit is not saved automatically and it tries to access the file at the path `scorm/block_id/hash_of_file` but that directory doesn't exist as nothing was saved there.

**Possible solutions**

- The simplest solution would be to just never delete the directory when the course is imported. However, I do not know if there is a way to check the course was imported. This does not resolve the subsection/unit duplicate issue.
- Save the `package_file` onto the storage as was already being done before [this commit](https://github.com/overhangio/openedx-scorm-xblock/commit/12ef19abd3077873654ee5ab604d69a8f2420a35) and extract the zip_file contents again onto the storage in case the original directory does not exist on the storage.
- Change the location where scorm stores the zip_file contents to make it unique for each course. For example appending course_id to block_id to make it something like `scorm/block_id+course_id/hash_of_file`. This would make separate directories for the exported and the imported course. However, this would require the zip_file contents to be extracted again as there would be nothing saved at the new directory resulting in the same issue as the subsection/unit duplicate.

### PR DESCRIPTION
**Pull request by Danyal-Faheem**
Fixes #70.

**Changes**

- Added a new base_path for the scorm folder. The previous one was using the block_id as an identifier for the directory name but when exporting a course and importing it, the block_id stayed the same causing both the courses to share the scorm directory. Now, the base_path is a sha1 hash of the usage_key which is unique across different courses as it also takes into account the course_key as well.
- The previous base_path is still kept for backwards compatibility and that path is returned in case storage exists on that path, meaning that the scorm module was created before this change.

**Caveats**

- If a scorm unit has already been initialized, i.e, scorm file has been uploaded, then duplicating that subsection or exporting and importing that course will still break the scorm module as earlier due to the path not being updated to the new version.
- Due to above point, this change will only effect the scorm modules initialized after this PR has been merged. Any course already created will still be facing the issue mentioned in #70 until the course author reupload scorm files and save again, i.e `studio_submit()` is called again.

### CODE REVIEW

**Review by ziafazal — COMMENTED** [review]
@Danyal-Faheem I think we should avoid uploading of content file it does not make sense to extract old package when course is imported. I think in case of course import scorm package should become unavailable in new course whereas in previous course it should work fine.

File: openedxscorm/scormxblock.py [hunk 1]
```diff
@@ -300,6 +319,17 @@ def studio_submit(self, request, _suffix):
         package_file = request.params["file"].file
         self.update_package_meta(package_file)
 
+        # First, save scorm file in the storage for later use
```
[the remark this answers is not in the document]
Comment (Danyal-Faheem) [reply openedxscorm/scormxblock.py:322]: This code was for uploading and saving the content file at the storage backend. However, if we don't need to save the file this isn't required anymore.

**Comment by Danyal-Faheem** [comment]
> @Danyal-Faheem I think we should avoid uploading of content file it does not make sense to extract old package when course is imported. I think in case of course import scorm package should become unavailable in new course whereas in previous course it should work fine.

@ziafazal okay, I will remove the code for saving the content file and just fix the path then.

**Review by ziafazal — APPROVED** [review]
👍

**Pull request merged**